diff --git a/tau/tau.h b/tau/tau.h
--- a/tau/tau.h
+++ b/tau/tau.h
Assertion failed: hunk, file ../patch-2.5.9-src/patch.c, line 354  
@@ -250,7 +250,7 @@ static inline double tauClock() {
     LARGE_INTEGER frequency;
     QueryPerformanceCounter(&counter);
     QueryPerformanceFrequency(&frequency);
-    return TAU_CAST(double, (counter.QuadPart * 1000 * 1000 * 1000) / frequency.QuadPart); // in nanoseconds
+    return TAU_CAST(double, (counter.QuadPart * 1000 * 1000 * 1000)) / TAU_CAST(double, frequency.QuadPart); // in nanoseconds
 
 #elif defined(__linux) && defined(__STRICT_ANSI__)
     return TAU_CAST(double, clock()) * 1000000000 / CLOCKS_PER_SEC; // in nanoseconds
@@ -446,17 +446,13 @@ static inline int tauIsDigit(const char c) { return c >= '0' && c <= '9'; }
 static inline int tauShouldDecomposeMacro(const char* const actual, const char* const expected, const int isStringCmp) {
     // Signal that the macro can be further decomposed if either of the following symbols are present
     int dots = 0;
-    int numActualDigits = 0;
-    int numExpectedDigits = 0;
 
     // If not inside a string comparison, we will return `1` only if we determine that `actual` is a variable
     // name/expression (i.e for a value, we search through each character verifying that each is a digit
     // - for floats, we allow a maximum of 1 '.' char)
     if(!isStringCmp) {
         for(int i = 0; i < strlen(actual); i++) {
-            if(tauIsDigit(actual[i])) {
-                numActualDigits++;
-            } else if(actual[i] == '.') {
+            if(actual[i] == '.') {
                 dots++;
                 if(dots > 1) { return 1; }
             } else {
@@ -465,10 +461,8 @@ static inline int tauShouldDecomposeMacro(const char* const actual, const char*
         }
         // Do the same for `expected`
         dots = 0;
-        for(int i=0; i < strlen(expected); i++) {
-            if(tauIsDigit(expected[i])) {
-                numExpectedDigits++;
-            } else if(expected[i] == '.') {
+        for(size_t i=0; i < strlen(expected); i++) {
+            if(expected[i] == '.') {
                 dots++;
                 if(dots > 1) { return 1; }
             } else {
diff --git a/tau/types.h b/tau/types.h
index b09c241..52e39ab 100644
--- a/tau/types.h
+++ b/tau/types.h
@@ -87,8 +87,8 @@ typedef tau_i32 tau_bool32; // Prefer this!
 // (U)Intptr is only here for semantic reasons really as this library will only support 32/64 bit OSes.
 // Are there any modern OSes (not 16 bit) where tau_iptr != ptrdiff_t/tau_ll ?
 #if defined(_WIN64)
-    typedef signed    __int64    tau_iptr;
-    typedef unsigned  __int64    tau_uptr;
+    typedef int64_t    tau_iptr;
+    typedef uint64_t   tau_uptr;
 #elif defined(_WIN32) && (defined(__MINGW32__) || defined(__MINGW64__))
     // MinGW: Use standard types to match tau_ll/tau_ull
     typedef tau_i64    tau_iptr;
