diff --git a/tau/tau.h b/tau/tau.h
index ab0ef17..bd2f3df 100644
--- a/tau/tau.h
+++ b/tau/tau.h
@@ -446,17 +446,13 @@ static inline int tauIsDigit(const char c) { return c >= '0' && c <= '9'; }
 static inline int tauShouldDecomposeMacro(const char* const actual, const char* const expected, const int isStringCmp) {
     // Signal that the macro can be further decomposed if either of the following symbols are present
     int dots = 0;
-    int numActualDigits = 0;
-    int numExpectedDigits = 0;
 
     // If not inside a string comparison, we will return `1` only if we determine that `actual` is a variable
     // name/expression (i.e for a value, we search through each character verifying that each is a digit
     // - for floats, we allow a maximum of 1 '.' char)
     if(!isStringCmp) {
         for(int i = 0; i < strlen(actual); i++) {
-            if(tauIsDigit(actual[i])) {
-                numActualDigits++;
-            } else if(actual[i] == '.') {
+            if(actual[i] == '.') {
                 dots++;
                 if(dots > 1) { return 1; }
             } else {
@@ -466,9 +462,7 @@ static inline int tauShouldDecomposeMacro(const char* const actual, const char*
         // Do the same for `expected`
         dots = 0;
         for(int i=0; i < strlen(expected); i++) {
-            if(tauIsDigit(expected[i])) {
-                numExpectedDigits++;
-            } else if(expected[i] == '.') {
+            if(expected[i] == '.') {
                 dots++;
                 if(dots > 1) { return 1; }
             } else {
